cpp/unit/ability.h

```diff
@@ -1,28 +1,53 @@
-// Copyright 2014-2014 the openage authors. See copying.md for legal info.
+// Copyright 2014-2015 the openage authors. See copying.md for legal info.
 
 #ifndef OPENAGE_UNIT_ABILITY_H_
 #define OPENAGE_UNIT_ABILITY_H_
 
 #include <bitset>
 #include <memory>
+#include <type_traits>
+#include <unordered_map>
 
 #include "../coord/phys3.h"
+#include "resource.h"
 
 namespace openage {
 
-class UnitAction;
-class Texture;
-class TestSound;
+class Command;
+class Sound;
 class Unit;
+class UnitAction;
+class UnitProducer;
+class UnitTexture;
 
+/**
+ * roughly the same as command_ability in game data
+ */
 enum ability_type {
 	move,
+	garrison,
+	ungarrison,
 	patrol,
+	train,
+	build,
+	research,
 	gather,
-	attack
+	attack,
+	convert,
+	repair,
+	heal
+};
+
+std::string to_string(const ability_type &at);
+
+struct ability_hash {
+	size_t operator ()(const ability_type &t) const {
+		return t;
+	}
```

Comment:
Same thing here, you should really avoid std if you can.

If you are interested in a very good hash library interface  ( not std : ) or if you are just curious see : 
http://m.youtube.com/watch?v=Njjp_MJsgt8

If we intend to have a lot of hashable types it could be interesting to have something similar.